src/components/HeadCellMenu/HeadCellMenu.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useState, useCallback, useEffect } from 'react';
+import Menu from '@qlik-trial/sprout/icons/Menu';
+import useFieldSelection from './use-field-selection';
+import RecursiveMenuList from './MenuList/RecursiveMenuList';
+import { getMenuItemGroups } from './utils';
+import { HeadCellMenuWrapper, StyledMenuButton } from './styles';
+import {
+  AdjustColumnSizeRelatedArgs,
+  FlagsArgs,
+  HeadCellMenuProps,
+  SearchRelatedArgs,
+  SelectionRelatedArgs,
+  SortingRelatedArgs,
+} from './types';
+
+const HeadCellMenu = <T extends HeadCellMenuProps>({
+  column,
+  tabIndex,
+  anchorRef,
+  translator,
+  handleHeadCellMenuKeyDown,
+  menuAvailabilityFlags,
+
+  // sorting
+  sortFromMenu,
+  changeActivelySortedColumn,
+  // search
+  embed,
+  listboxRef,
+  interactions,
+  // selection
+  app,
+  model,
+  // adjust col size
+  setFocusOnClosetColumnAdjuster,
+}: T &
+  FlagsArgs<T['menuAvailabilityFlags']> &
+  SortingRelatedArgs &
+  SearchRelatedArgs &
+  SelectionRelatedArgs &
+  AdjustColumnSizeRelatedArgs) => {
+  const { headTextAlign, qLibraryId, fieldId } = column;
+  const [openMenuDropdown, setOpenMenuDropdown] = useState(false);
+  const {
+    fieldInstance,
+    selectionActionsEnabledStatus,
+    resetSelectionActionsEnabledStatus,
+    updateSelectionActionsEnabledStatus,
+  } = useFieldSelection({ column, app });
+
+  const handleOpenDropdown = async () => {
+    if (!openMenuDropdown && model) {
+      const layout = await model.getLayout();
+      updateSelectionActionsEnabledStatus(layout as EngineAPI.IGenericHyperCubeLayout);
+    }
+    setOpenMenuDropdown(!openMenuDropdown);
+  };
+
+  const embedListbox = useCallback(() => {
+    const id = qLibraryId ? { qLibraryId, type: 'dimension' } : fieldId;
+    // @ts-expect-error TODO: no types for `__DO_NOT_USE__`, it will improve when it becomes stable
+    embed.__DO_NOT_USE__.popover(listboxRef.current, id, {
+      anchorOrigin: { vertical: 'bottom', horizontal: 'left' },
+      transformOrigin: { vertical: 'top', horizontal: 'left' },
+    });
+  }, [embed, fieldId, qLibraryId, listboxRef]);
+
+  useEffect(() => {
+    if (!openMenuDropdown) resetSelectionActionsEnabledStatus();
+  }, [openMenuDropdown, resetSelectionActionsEnabledStatus]);
+
+  return (
+    <HeadCellMenuWrapper rightAligned={headTextAlign === 'right'}>
+      <StyledMenuButton
+        size="small"
+        tabIndex={tabIndex}
+        id="sn-table-head-menu-button"
+        aria-controls={openMenuDropdown ? 'sn-table-head-menu' : undefined}
```

Comment:
Not really sure what we want to do with `sn-table` prefixes. But my first thought is that we should use something that is either generic (works for both charts) or explicit that is defined (ex. set as a property of the component).

Reply:
> Not really sure what we want to do with `sn-table` prefixes.

**in case of translations:**
I think [this is the solution here](https://github.com/qlik-oss/nebula-table-utils/blob/1132b21f2e6be595dd13946bf1f320240ac0d644/src/locale/locales/en.json) for us in this case, My basic idea is, if some string ends up in this repo, definitely will have it's own translation originated from this repo. 

**in case of selector identifiers** (like this one you mentioned):
we should have all selectors originated from here as well and i'll fix it soon